### PR TITLE
planner: update seek-factor in cost model ver2

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #35240

Problem Summary: planner: update seek-factor in cost model ver2

### What is changed and how it works?

1. output cost factor weights under `true-card-cost` mode for cost calibration;
2. since the number of requests is tough to estimate, for simplicity and robustness, only let operators that need double-read like IndexLookup and IndexJoin consider this cost.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Reminiscent — APPROVED** [review]

File: planner/core/plan_cost_ver2.go [hunk 1]
```diff
@@ -212,12 +206,8 @@ func (p *PhysicalTableReader) getPlanCostVer2(taskType property.TaskType, option
 
 // getPlanCostVer2 returns the plan-cost of this sub-plan, which is:
 // plan-cost = index-side-cost + (table-side-cost + double-read-cost) / double-read-concurrency
-// index-side-cost = (index-child-cost + index-net-cost + index-seek-cost) / dist-concurrency # same with IndexReader
-// table-side-cost = (table-child-cost + table-net-cost + table-seek-cost) / dist-concurrency # same with TableReader
-// double-read-cost = double-read-seek-cost + double-read-cpu-cost
-// double-read-seek-cost = double-read-tasks * seek-factor
-// double-read-cpu-cost = index-rows * cpu-factor
-// double-read-tasks = index-rows / batch-size * task-per-batch # task-per-batch is a magic number now
```
Comment (xuyifangreeneyes) [inline planner/core/plan_cost_ver2.go:220]: Why do we remove the comments? It seems that we only need to remove `index-seek-cost` and `table-seek-cost`.

File: planner/core/plan_cost_ver2.go [hunk 2]
```diff
@@ -268,15 +256,12 @@ func (p *PhysicalIndexLookUpReader) getPlanCostVer2(taskType property.TaskType,
 
 // getPlanCostVer2 returns the plan-cost of this sub-plan, which is:
 // plan-cost = table-side-cost + sum(index-side-cost)
-// index-side-cost = (index-child-cost + index-net-cost + index-seek-cost) / dist-concurrency # same with IndexReader
-// table-side-cost = (table-child-cost + table-net-cost + table-seek-cost) / dist-concurrency # same with TableReader
```
Comment (xuyifangreeneyes) [inline planner/core/plan_cost_ver2.go:272]: ditto

File: planner/core/plan_cost_ver2.go (hunk 1 above)
Comment (qw4990) [reply planner/core/plan_cost_ver2.go:220]: I add them back, PTAL

**Review by xuyifangreeneyes — APPROVED** [review]

File: planner/core/plan_cost_ver2.go [hunk 3]
```diff
@@ -212,12 +206,14 @@ func (p *PhysicalTableReader) getPlanCostVer2(taskType property.TaskType, option
...
-// double-read-cost = double-read-seek-cost + double-read-cpu-cost
-// double-read-seek-cost = double-read-tasks * seek-factor
+// index-side-cost = (index-child-cost + index-net-cost) / dist-concurrency # same with IndexReader
+// table-side-cost = (table-child-cost + table-net-cost) / dist-concurrency # same with TableReader
+// double-read-cost = double-read-request-cost + double-read-cpu-cost
+// double-read-request-cost = double-read-tasks * request-factor
 // double-read-cpu-cost = index-rows * cpu-factor
 // double-read-tasks = index-rows / batch-size * task-per-batch # task-per-batch is a magic number now
+// index-side-cost = (index-child-cost + index-net-cost) / dist-concurrency # same with IndexReader
+// table-side-cost = (table-child-cost + table-net-cost) / dist-concurrency # same with TableReader
```
[the remark this answers is not in the document]
Comment (qw4990) [reply planner/core/plan_cost_ver2.go]: updated

**Pull request merged**